pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -522,7 +522,7 @@ export async function resolveDependencies (
       postponedPeersResolutionQueue.push(postponedPeersResolution)
     }
   })
-  const newPreferredVersions = { ...preferredVersions }
+  const newPreferredVersions = Object.create(preferredVersions) as PreferredVersions
```

Comment:
But this means that when we replace a field in `newPreferredVersions`, the same field will also be replaced in `preferredVersions`, which is not what we want.

Reply:
hm, looks like in this case we don't override existing fields in `newPreferredVersions`, only adding new ones. And the new ones will not appear in `preferredVersions`. So I guess it should work fine.

Reply:
there is no change in this regard, when assigning the property to `newPreferredVersions` - it would affect only the new object created here (not a prototype of it)

the main difference is to be when reading the property - if it not found in `newPreferredVersions`, it would be read from the prototype (`preferredVersions`)

however, the opposite is true - changing properties of `preferredVersions`, if not overridden will be reflected on `newPreferredVersions`

Reply:
about the possibility of overrides in `preferredVersions`, just noticed in code below(for both - before and after the pr), not sure if that can cause issues:

```js
    if (!newPreferredVersions[resolvedPackage.name]) {
      newPreferredVersions[resolvedPackage.name] = {}
    }
    if (!newPreferredVersions[resolvedPackage.name][resolvedPackage.version]) {
      newPreferredVersions[resolvedPackage.name][resolvedPackage.version] = 'version'
      // ^^ -- here the content of `preferredVersions[resolvedPackage.name][resolvedPackage.version]` is possibly modified
      // that will happen if `resolvedPackage.name` was already in `preferredVersions` ()
    }
```

Reply:
yes, I think this should be fixed. It is probably causing some hard to reproduce issues.

Reply:
I have created an issue from your comment https://github.com/pnpm/pnpm/issues/6737

Let me know if you want to work on it.